metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -59,6 +59,7 @@ type enricher struct {
 	watcher            kubernetes.Watcher
 	watcherStarted     bool
 	watcherStartedLock sync.Mutex
+	isPod              bool
```

Comment:
Instead of having conditional logic we could maybe have an specialized enricher for pods.